### ISSUE TITLE
Turkish capitalization rules cause issues with compiler, widget creation

### ISSUE THREAD

**Issue by nicolaspayette** [NetLogo/NetLogo#317]
NetLogo keywords involving the letter `i` cause compilation errors, e.g.:

```
Nothing named THİCKNESS has been defined
```

when writing `thickness` in lowercase.

A recent user email brought the issue forward again, but it had been reported before:

http://groups.yahoo.com/group/netlogo-users/message/7798
http://groups.yahoo.com/group/netlogo-users/message/15356

The basic issue is that a lowercase `i` doesn't capitalize to a regular uppercase `I` when using the Turkish locale:

```
scala> "i".toUpperCase(new java.util.Locale("tr"))
res7: java.lang.String = İ
```

To reproduce, just start NetLogo with `-Duser.language=tr`.

To workaround, start NetLogo with `-Duser.language=en` (this can be added to `NetLogo.sh`) or manually uppercase all problematic names in your NetLogo code (i.e., use `THICKNESS` instead of `thickness`).

A permanent fix involves explicitly using `.toUpperCase(java.util.Locale.ENGLISH)` in places where it matters.

**Comment by mrerrormessage** [comment]
Quick update on the workaround - another user ran into this problem recently. The workaround should still be good, but it needs to be applied somewhat differently. NetLogo no longer has a .vmoptions file, instead it now has a cfg file. Instructions on where to find this file are [here](http://ccl.northwestern.edu/netlogo/docs/faq.html#howbig). Add the following two lines at the end of the `[JVMOptions]` section (you don't need to add the make the changes described in the FAQ). 

```
-Duser.language=en
-Duser.country=US
```

**Comment by alan-isaac** [comment]
Could this issue explain why a Turkish student of mine found NetLogo did not recognize the `extensions` keyword?  Our workaround was to create an English-language account for her.  (NetLogo 6.0.2 on Win 10.)

**Comment by mrerrormessage** [comment]
Yes, this would almost certainly explain why the `extensions` keyword will not work for a student with a Turkish locale. I'm not certain that a separate account is necessary, the workaround described above (editing the NetLogo.cfg file) should work for the student.
